src/nnue/nnue_accumulator.cpp

```diff
@@ -226,171 +226,153 @@ void update_accumulator_incremental(
     else
         FeatureSet::append_changed_indices<Perspective>(ksq, computed.dirtyPiece, added, removed);
 
-    if (removed.size() == 0 && added.size() == 0)
+    assert(added.size() == 1 || added.size() == 2);
+    assert(removed.size() == 1 || removed.size() == 2);
+
+    if (Forward)
+        assert(added.size() <= removed.size());
+    else
+        assert(removed.size() <= added.size());
+
+#ifdef VECTOR
+    auto* accIn  = reinterpret_cast<const vec_t*>(&(computed.*accPtr).accumulation[Perspective][0]);
+    auto* accOut = reinterpret_cast<vec_t*>(&(target_state.*accPtr).accumulation[Perspective][0]);
+
+    const IndexType offsetA0 = TransformedFeatureDimensions * added[0];
+    auto* columnA0 = reinterpret_cast<const vec_t*>(&featureTransformer.weights[offsetA0]);
+    const IndexType offsetR0 = TransformedFeatureDimensions * removed[0];
+    auto* columnR0 = reinterpret_cast<const vec_t*>(&featureTransformer.weights[offsetR0]);
+
+    if ((Forward && removed.size() == 1) || (Backwards && added.size() == 1))
+    {
+        assert(added.size() == 1 && removed.size() == 1);
+        for (IndexType i = 0; i < TransformedFeatureDimensions * sizeof(WeightType) / sizeof(vec_t);
+             ++i)
+            accOut[i] = vec_add_16(vec_sub_16(accIn[i], columnR0[i]), columnA0[i]);
+    }
+    else if (Forward && added.size() == 1)
+    {
+        assert(removed.size() == 2);
+        const IndexType offsetR1 = TransformedFeatureDimensions * removed[1];
+        auto* columnR1 = reinterpret_cast<const vec_t*>(&featureTransformer.weights[offsetR1]);
+
+        for (IndexType i = 0; i < TransformedFeatureDimensions * sizeof(WeightType) / sizeof(vec_t);
+             ++i)
+            accOut[i] =
+              vec_sub_16(vec_add_16(accIn[i], columnA0[i]), vec_add_16(columnR0[i], columnR1[i]));
+    }
+    else if (Backwards && removed.size() == 1)
     {
-        std::memcpy((target_state.*accPtr).accumulation[Perspective],
```

Comment:
This seems unexpected. Do we understand what specific change made this code not get used?  I just want to make sure some less optimal path isn't being used instead.